### PR TITLE
SDK-1341: Feature/get node favourite

### PR DESCRIPTION
**Pull request by ms-mega**

### CODE REVIEW

**Comment by ms-mega** [comment]
trigger compilation

**Comment by ms-mega** [comment]
trigger compilation

**Comment by ms-mega** [comment]
trigger compilation

File: include/megaapi.h [hunk 1]
```diff
@@ -2996,7 +2996,7 @@ class MegaRequest
             TYPE_CREDIT_CARD_CANCEL_SUBSCRIPTIONS, TYPE_GET_SESSION_TRANSFER_URL,
             TYPE_GET_PAYMENT_METHODS, TYPE_INVITE_CONTACT, TYPE_REPLY_CONTACT_REQUEST,
             TYPE_SUBMIT_FEEDBACK, TYPE_SEND_EVENT, TYPE_CLEAN_RUBBISH_BIN,
-            TYPE_SET_ATTR_NODE, TYPE_CHAT_CREATE, TYPE_CHAT_FETCH, TYPE_CHAT_INVITE,
+            TYPE_SET_ATTR_NODE, TYPE_GET_ATTR_NODE, TYPE_CHAT_CREATE, TYPE_CHAT_FETCH, TYPE_CHAT_INVITE,
```
[the remark this answers is not in the document]
Comment (sergiohs84) [reply include/megaapi.h:2999]: Also, for any new type of request, we add it to the `MegaRequestPrivate::getRequestString()`

File: include/megaapi.h [hunk 2]
```diff
@@ -10953,6 +10966,15 @@ class MegaApi
          */
         void setNodeFavourite(MegaNode *node, bool fav, MegaRequestListener *listener = NULL);
 
+        /**
+         * @brief Get a list of favourite nodes.
+         *
```
Comment (sergiohs84) [inline include/megaapi.h:10971]: The information about the type of request and the valid members is missing. See others for reference, please.

**Review by sergiohs84 — CHANGES_REQUESTED** [review]

File: tests/integration/SdkTest_test.cpp [hunk 3]
```diff
@@ -644,6 +644,13 @@ void SdkTest::onRequestFinish(MegaApi *api, MegaRequest *request, MegaError *e)
     case MegaRequest::TYPE_FETCH_GOOGLE_ADS:
         mApi[apiIndex].mStringMap.reset(mApi[apiIndex].lastError == API_OK ? request->getMegaStringMap()->copy() : nullptr);
             break;
+
+    case MegaRequest::TYPE_GET_ATTR_NODE:
+        if (mApi[apiIndex].lastError == API_OK)
+        {
+            mMegaFavNodeList = request->getMegaNodeList()->copy();
```
Comment (sergiohs84) [inline tests/integration/SdkTest_test.cpp:651]: You are accumulating a memory leak here, every time you make a copy, the previous allocated memory is leaked.

**Review by sergiohs84 — CHANGES_REQUESTED** [review]

File: tests/integration/SdkTest_test.cpp (hunk 3 above)
Comment (sergiohs84) [reply tests/integration/SdkTest_test.cpp]: Fixed in 8f5d15a

**Review by sergiohs84 — APPROVED** [review]

**Pull request merged**